examples/rpc/partner-plugin/handlers/accommodation/v1/accommodation_search.go

```diff
@@ -0,0 +1,271 @@
+					exists = true
+					break
+				}
+			}
+			if !exists {
+				available_properties = append(available_properties, prop)
+			}
+		}
+
+		var price = 2
+
+		// generate search result
+		for _, prop := range available_properties {
+
+			// units requested
+			units_requested := query.UnitCount
+
+			// empty units array
+			units := make([]*accommodationv1.Unit, 0)
+
+			// loop all rooms
+			for _, room := range prop.Rooms {
+
+				units = append(units, &accommodationv1.Unit{
+					Type:             0,
+					SupplierRoomCode: room.SupplierCode,
+					SupplierRoomName: room.SupplierName,
+					OriginalRoomName: room.OriginalName,
+					TravelPeriod: &typesv1.TravelPeriod{
+						StartDate: &typesv1.Date{
+							Year:  query.TravelPeriod.GetStartDate().GetYear(),
+							Month: query.TravelPeriod.GetStartDate().GetMonth(),
+							Day:   query.TravelPeriod.GetStartDate().GetDay(),
+						},
+						EndDate: &typesv1.Date{
+							Year:  query.TravelPeriod.GetEndDate().GetYear(),
+							Month: query.TravelPeriod.GetEndDate().GetMonth(),
+							Day:   query.TravelPeriod.GetEndDate().GetDay(),
+						},
+					},
+					TravellerIds: getTravellerIds(query.Travellers),
+					Beds:         room.Beds,
+					PriceDetail: &typesv1.PriceDetail{
+						Price: &typesv1.Price{
+							Value: fmt.Sprintf("%d", price),
+							Currency: &typesv1.Currency{
+								Currency: req.SearchParametersGeneric.Currency.Currency,
+							},
+						},
+					},
+					Services:       []*typesv1.ServiceFact{},
+					MealPlanCode:   &typesv1.MealPlan{},
+					RatePlan:       &typesv1.RatePlan{},
+					RateRule:       &typesv1.RateRule{},
+					CancelPolicies: []*typesv1.CancelPolicy{},
+					RemainingUnits: 0,
+					PropertyCode:   &typesv1.ProductCode{},
+					SupplierCode:   prop.Property.SupplierCode,
+					Remarks:        "",
+				})
+
+				price += 2
+
+				if units_requested == int32(len(units)) {
+					break
+				}
+			}
+
+			// check how many units are requested
+			if units_requested == int32(len(units)) {
+				searchResults = append(searchResults, &accommodationv1.AccommodationSearchResult{
+					ResultId: int32(len(searchResults) + 1),
+					QueryId:  query.QueryId,
+					TotalPriceDetail: &typesv1.PriceDetail{
+						Price: &typesv1.Price{
+							Value: fmt.Sprintf("%d", price),
+							Currency: &typesv1.Currency{
+								Currency: req.SearchParametersGeneric.Currency.Currency,
+							},
+						},
+					},
+					Units: units,
+				})
+			}
+		}
+	}
+
+	if len(searchResults) == 0 {
+		return &accommodationv1.AccommodationSearchResponse{
+			Header: &typesv1.ResponseHeader{
+				Status: typesv1.StatusType_STATUS_TYPE_SUCCESS,
+				Alerts: []*typesv1.Alert{
+					{
+						Message: fmt.Sprintf("No results found for search %v", req.Queries),
+						Type:    typesv1.AlertType_ALERT_TYPE_INFO,
+					},
+				},
+			},
+		}, nil
+	}
+
+	// generate a random string of 8 numbers
+	cache := cache.NewSearchCache()
```

Comment:
Is there a reason why a search cache has been implemented?
IIRC the discussion we had was about simple static data and in case we have multiple search ids we just have then multiple verify/mint ids reflecting them. In this case a caching would be unnecessary.

Reply:
cache service removed.